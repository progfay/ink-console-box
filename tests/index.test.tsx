import React from 'react'
import { render } from 'ink-testing-library'
import { ConsoleBox } from '../lib'

const waitCall = (): { callback: () => void, called: Promise<void> } => {
  let callback: () => void = () => {}
  const called: Promise<void> = new Promise(resolve => { callback = resolve })
  return { callback, called }
}

describe('<ConsoleBox>', () => {
  it('Single line', async () => {
    const { callback, called } = waitCall()

    const { lastFrame } = render(
      <ConsoleBox
        func={console => {
          console.log(12345)
        }}
        onSuccess={callback}
        onFailed={callback}
      />
    )

    await called
    expect(lastFrame()).toMatchSnapshot()
  })

  it('Multi line', async () => {
    const { callback, called } = waitCall()

    const { lastFrame } = render(
      <ConsoleBox
        func={console => {
          console.log(12345)
          console.log(12345)
        }}
        onSuccess={callback}
        onFailed={callback}
      />
    )

    await called
    expect(lastFrame()).toMatchSnapshot()
  })

  it('Error', async () => {
    const { callback, called } = waitCall()

    const { lastFrame } = render(
      <ConsoleBox
        func={console => {
          throw Error('error!')
        }}
        onSuccess={callback}
        onFailed={callback}
      />
    )

    await called
    expect(lastFrame()).toMatchSnapshot()
  })
})
