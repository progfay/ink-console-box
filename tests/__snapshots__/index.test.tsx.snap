// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsoleBox> Error 1`] = `"[1m[90m 1 [22m[39m[90mâ”‚[39m Error: error!"`;

exports[`<ConsoleBox> Multi line 1`] = `
"[1m[90m 1 [22m[39m[90mâ”‚[39m [33m12345[39m
[1m[90m 2 [22m[39m[90mâ”‚[39m [33m12345[39m"
`;

exports[`<ConsoleBox> Single line 1`] = `"[1m[90m 1 [22m[39m[90mâ”‚[39m [33m12345[39m"`;
